app/models/street/StreetEdgeTable.scala

```diff
@@ -139,17 +222,50 @@ object StreetEdgeTable {
     * @param auditCount
     * @return
     */
-  def auditedStreetDistance(auditCount: Int): Float = db.withSession { implicit session =>
+  def auditedStreetDistance(auditCount: Int, auditTaskQuery: Query[AuditTaskTable, AuditTask, Seq] = completedAuditTasks): Float = db.withSession { implicit session =>
```

Comment:
So I generally think that it is bad practice to pass queries around as parameters. Luckily, there is a better way to do this :)

You could do the same thing here as I did in that bit of code that I rewrote to use the `match` syntax. It would look something like
```
def auditedStreetDistance(auditCount: Int, userType: String = "All"): Float = db.withSession { implicit session =>
  val auditTaskQuery = userType match {
    case "All" => completedAuditTasks
    case "Researcher" => researcherCompletedAuditTasks
    ...
```

So you would only have to take in the String that describes the user type, and you grab the appropriate set of audit tasks based on that String.

You should be able to delete half the lines of code that you've written once you've done that, because `streetDistanceCompletionRate*` and `auditCompletionRate*` can be replaced by just taking in that String, and passing it along to this `auditedStreetDistance` function. So you only need one of each. And you don't even need to check what is in the String until you get to `auditedStreetDistance`. So you can delete the example code I wrote in `auditCompletionRate`, and just pass the String right to this function instead.